lib/presentation/settings_page.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_mentoring/presentation/localization/locale_inherited_widget.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import 'utils/extensions.dart';
+
+class SettingsPage extends StatelessWidget {
+  static final route = "/settings";
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(context.getLocalizedString("settings_title")),
+      ),
+      body: Row(
+        children: [
+          Text(context.getLocalizedString("settings_language_label")),
+          LanguagePickerWidget(),
+        ],
+      ),
+    );
+  }
+}
+
+class LanguagePickerWidget extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    var localeInheritedWidget = LocaleInheritedWidget.of(context);
+
+    return FutureBuilder<SharedPreferences>(
+        future: SharedPreferences.getInstance(),
```

Comment:
This is wrong usage of FutureBuilder not even taking the usage of SharedPreferences on UI into account :)

In your case this future will be created and calculated each time LanguagePickerWidget#build is called. So if it was a network call, it would be executed each LanguagePickerWidget#build call (it can be called maaaany maaaaaany times while your app is running). You can instantiate this future outside build method (make LanguagePickerWidget stateful and instantiate the future in initState of pass the future from without of this widget), but it will still not resolve your bug here.
When you make the LocaleInheritedWidget implemented right you should refer it's data instead of getting it from SharedPreferences.

Reply:
That is interesting! =)

1) Supposing SharedPreferences is used correct (just to better understanding your point about FutureBuilder). For creating the future is better to use `initState`, right? Because `build` method is called very often for UI needs, but `initState` just when screen is opened on goes foreground? Should it look like that?
```kotlin

class _LanguagePickerWidgetState extends State<LanguagePickerWidget> {
  late Future<SharedPreferences> sharedPrefsFuture;

  @override
  void initState() {
    super.initState()
    sharedPrefsFuture = SharedPreferences.getInstance();
  }

  @override
  Widget build(BuildContext context) {
    var localeInheritedWidget = LocaleInheritedWidget.of(context);

    return FutureBuilder<SharedPreferences>(
        future: sharedPrefsFuture,
// the rest of code
```
2) Could you elaborate on how is it better to use shared preferences in this case? You mean that I have to create some wrapper for it? But I will still need to call it async every time so that have updated values and I cannot do it in build method, only with FutureBuilder...